design/apply-config-from-cli.md

```diff
@@ -0,0 +1,68 @@
+# Allow applying application configuration from CLI as a stand-alone operation
+
+Admin Console CLI has a way to specify appication configuration during the initial install.
+However there is no dedicated way to use CLI to update application configuration.
+`kots download` and `kots upload` commands can be used to achieve this goal.
+However, this does not allow to easily patch the config without knowing the current set of keys and values.
+
+## Goals
+
+- Configuration can be applied independently from the rest of the app spec (unlike kots download/upload commands).
+- Configuration can be applied to new versions of the app before they are deployed.
+- Individual config keys can be added, modified, and removed.
+
+## Non Goals
+
+- Changing current architecture, in which config changes are always separate from other changes, such as license updates or application version updates.
+License updates and upstream updates will still create their own versions with existing config.
+- Covering every add/delete/modify config key cases in convenience commands, though the CLI can be extended later.
+- Interractive editing in terminal.
+- Adding the ability to make config changes to a deployed application without having to also take the latest downloaded upstream app version. 
+
+## Background
+
+Some application require periodic configuration updates.
+CLI will provide a way to automate such operations.
+
+## High-Level Design
+
+There will be two modes of operation:
+1. Apply an entire configuration to the app.
+1. Apply a config change to a subset of values.  This is mostly a convenience method since this can be achieved with the first mode.
+
+## Detailed Design
+
+The following command will be added to CLI:
+
+```
+kubectl kots config <app-slug> set [parameters]
```

Comment:
Why is `set` needed here? Is that because we're leaving room for a `get` and `export`?

It conflicts with my use of `--set` below, but if we do keep set, then I'd suggest we should simply support key=value pairs after the `set` command without having to use `--` args at all.

Lastly, should we follow k8s verb-noun and use `kubectl kots set config`?

Reply:
> Why is set needed here? Is that because we're leaving room for a get and export?
yes
> Lastly, should we follow k8s verb-noun and use kubectl kots set config
yes, we can do this